pkgs/development/libraries/flutter-engine/default.nix

```diff
@@ -0,0 +1,283 @@
+            "src/buildtools/linux-x64/clang/bin/llvm-mt"
+            "src/buildtools/linux-x64/clang/bin/llvm-nm"
+            "src/buildtools/linux-x64/clang/bin/llvm-objcopy"
+            "src/buildtools/linux-x64/clang/bin/llvm-objdump"
+            "src/buildtools/linux-x64/clang/bin/llvm-pdbutil"
+            "src/buildtools/linux-x64/clang/bin/llvm-profdata"
+            "src/buildtools/linux-x64/clang/bin/llvm-rc"
+            "src/buildtools/linux-x64/clang/bin/llvm-readobj"
+            "src/buildtools/linux-x64/clang/bin/llvm-size"
+            "src/buildtools/linux-x64/clang/bin/llvm-symbolizer"
+            "src/buildtools/linux-x64/clang/bin/llvm-undname"
+            "src/buildtools/linux-x64/clang/bin/llvm-xray"
+          ];
+
+          shellFixes = [
+            "src/third_party/dart/tools/sdks/dart-sdk/bin/dart2js"
+            "src/flutter/tools/gn"
+          ];
+
+          ## FIXME: instead of patching a bunch of binaries, maybe we copy from nixpkgs?
+          installPhase = ''
+            mkdir -p $out
+            cp -r -P --no-preserve=mode,ownership $buildroot $out/src
+            cp -r -P --no-preserve=mode,ownership $src $out/src/flutter
+            ${concatStringsSep "\n" (attrValues (mapAttrs (name: src: ''
+              if [[ -d ${src} ]]; then
+                mkdir -p $out/${name}
+                cp -r -P --no-preserve=mode,ownership ${src}/* $out/${name}
+                find ${src} -type f -name '.*' | xargs -I {} cp --no-preserve=mode,ownership {} $out/${name}
+                ${optionalString (builtins.hasAttr "rev" src) ''
+                  mkdir -p $out/${name}/.git/logs
+                  echo "${src.rev}" >$out/${name}/.git/logs/HEAD
+                ''}
+              else
+                cp -P --no-preserve=mode,ownership ${src} $out/${name}
+              fi
+            '') flutter-deps))}
+            touch $out/src/third_party/dart/build/config/gclient_args.gni
+            cp $gitrev $out/src/flutter/build/git_revision.py
+            for bin in $binaryFixes; do
+              chmod 0755 $out/$bin
+              patchelf --set-interpreter ${stdenv.cc.libc}/lib/${interpreter} $out/$bin
+            done
+            for sh in $shellFixes; do
+              chmod 0755 $out/$sh
+              patchShebangs $out/$sh
+            done
+            chmod +x $out/src/build/linux/sysroot_ld_path.sh
+            sed -i '1 s|^.*$|#!${gnumake}/bin/make -f|' $out/src/third_party/harfbuzz/src/update-unicode-tables.make
+            mkdir -p $out/src/build/linux/debian_sid_${toolchainArch}-sysroot
+            tar -xf $toolchain -C $out/src/build/linux/debian_sid_${toolchainArch}-sysroot
+            cd $out
+            mkdir -p $out/src/third_party/dart/.dart_tool
+            cp -r --no-preserve=ownership,mode $dartPackageConfig $out/src/third_party/dart/.dart_tool/package_config.json
+            python3 $out/src/third_party/dart/tools/generate_package_config.py
```

Comment:
add comment, something like "these scripts need patched binaries in $out"

Reply:
Done